src/main/java/io/boomerang/service/refactor/TaskServiceImpl.java

```diff
@@ -139,22 +139,6 @@ public void createTask(InternalTaskRequest request) {
     taskExecution.setFlowTaskStatus(TaskStatus.inProgress);
     taskExecution = taskActivityService.save(taskExecution);
 
-    // Synchronize status update since `createTask` can be invoked multiple times, depending on the
```

Comment:
As this code manages Tasks and not Activities, the Status changes for Activities should be managed at a higher level.
This is already done in FlowExectuionServiceImpl->executeWorkflowAsync (line 254)